tools/state-viewer/src/commands.rs

```diff
@@ -984,3 +999,79 @@ pub(crate) fn clear_cache(store: Store) {
     store_update.delete_all(DBCol::CachedContractCode);
     store_update.commit().unwrap();
 }
+
+#[cfg(test)]
+mod tests {
+    use near_chain::types::RuntimeAdapter;
+    use near_chain::ChainGenesis;
+    use near_chain_configs::Genesis;
+    use near_client::test_utils::TestEnv;
+    use near_crypto::{InMemorySigner, KeyFile, KeyType};
+    use near_epoch_manager::EpochManager;
+    use near_primitives::shard_layout::ShardUId;
+    use near_primitives::types::chunk_extra::ChunkExtra;
+    use near_primitives::types::AccountId;
+    use nearcore::config::Config;
+    use nearcore::config::GenesisExt;
+    use nearcore::{NearConfig, NightshadeRuntime};
+    use std::sync::Arc;
+
+    #[test]
+    /// Tests that getting the latest trie state actually gets the latest state.
+    /// Adds a transaction and waits for it to be included in a block.
+    /// Checks that the change of state caused by that transaction is visible to `load_trie()`.
+    fn test_latest_trie_state() {
+        near_o11y::testonly::init_test_logger();
+        let validators = vec!["test0".parse::<AccountId>().unwrap()];
+        let genesis = Genesis::test_sharded_new_version(validators, 1, vec![1]);
+        let chain_genesis = ChainGenesis::test();
+
+        let tmp_dir = tempfile::tempdir().unwrap();
+        let home_dir = tmp_dir.path();
+
+        let store = near_store::test_utils::create_test_store();
+        let epoch_manager = EpochManager::new_arc_handle(store.clone(), &genesis.config);
+        let runtime =
+            NightshadeRuntime::test(tmp_dir.path(), store.clone(), &genesis, epoch_manager.clone())
+                as Arc<dyn RuntimeAdapter>;
+
+        let stores = vec![store.clone()];
+        let epoch_managers = vec![epoch_manager];
+        let runtimes = vec![runtime];
```

Comment:
Not sure I see the value of defining these variables explicitly over inlining them into the call below, because it's quite clear that `vec![thing]` is `things` and this would allow us to trim the size of the test.